### DISCUSSION TITLE
Record incorrect changes for another user

### DISCUSSION THREAD

**Discussion by mohammad-reza-najafi** [livewire/livewire#3456] [7. Bugs (archived)] [answered]
Hi
I was using a component in the admin.
I was on the user edit page and I wanted to get the access level from one user, and on another page my profile as an admin was open, but after I got the access from another user, I saw that the access was taken from the admin that I am. .
I mean, I was making changes to another profile, but it worked for me
The part codes that I give to users in the admin panel
`
        <form wire:submit.prevent="updatePermissions">

            <div class="form-group form-row">
                <label for="inputEmail3" class="col-sm-2 control-label">انتخاب دسترسی جدید</label>
                <select wire:model.debounce="permissions" class="form-control text-danger" name="permissions[]" id="permissions" multiple>
                    @foreach( \App\Models\Permission::whereDoesntHave('users', function($q) use ($user){
                  $q->where('user_id', $user->id);})->get()  as $permission)
                        <option value="{{ $permission->id }}">{{ $permission->name }} - {{ $permission->label }}</option>
                    @endforeach
                </select>
                <button class="btn btn-info" type="submit">اپدیت</button>
            </div>
        </form>
`

Component class codes

`
class EditUserPermissions extends Component
{
    use AuthorizesRequests;
    public $user;
    public $permissions;
    public $roles;
    protected $rules=[
        'permissions'       => ['required','array'],
        'roles'             => ['required','array']
    ];
    public function mount(User $user)
    {
        $this->user=$user;
    }

    protected $listeners = [
      'update.role' => '$refresh',
      'update.permission' => '$refresh',
    ];

    public function deleteThisRoleFromThisUser(Role $role)
    {
        $role->users()->detach();
        $this->dispatchBrowserEvent('alert',
            ['type' => 'success',  'message' => 'مقام مورد نظر برای کاربر حذف شد']);
        $this->emit('update.role');

    }
    public function updateRoles()
    {
        if (! Gate::allows('set-user-role')){
            $this->dispatchBrowserEvent('alert',
                ['type' => 'success',  'message' => 'شما اجازه ی دادن نقش به کاربر را ندارید']);
            return ;
        }
        $this->user->roles()->attach($this->roles);

        $this->emit('update.role');

        $this->dispatchBrowserEvent('alert',
            ['type' => 'success',  'message' => 'دسترسی ها برای کاربر مورد نظر با موفقیت اعمال شد']);
    }

    public function deleteThisPermissionFromThisUser(Permission $permission)
    {
        $permission->users()->detach();
        $this->dispatchBrowserEvent('alert',
            ['type' => 'success',  'message' => 'دسترسی  مورد نظر برای کاربر حذف شد']);
        $this->emit('update.permission');
    }
    public function updatePermissions()
    {
        if (! Gate::allows('set-user-permission')){
            $this->dispatchBrowserEvent('alert',
                ['type' => 'success',  'message' => 'شما اجازه دادن دسترسی به کاربر را ندارید']);
            return ;
        }
        $this->user->permissions()->attach($this->permissions);
        $this->emit('update.permission');

        $this->dispatchBrowserEvent('alert',
            ['type' => 'success',  'message' => 'مقام  ها برای کاربر مورد نظر با موفقیت اعمال شد']);
    }
    public function render()
    {
        return view('livewire.admin.users.edit-user-permissions');
    }
}

`

**Comment by mohammad-reza-najafi** [comment, marked as answer]
There was a problem with my code.
 When I got access, I got all the users who had this access

**Reply by joshhanley** [reply]
@mohammad-reza-najafi so your problem is fixed?

**Reply by mohammad-reza-najafi** [reply]
@joshhanley  yes it fixed

**Reply by joshhanley** [reply]
@mohammad-reza-najafi ok great 🙂
